src/Core/Resolvers/Factories/QueryEngineFactory.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.DataApiBuilder.Auth;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Core.Models;
+using Azure.DataApiBuilder.Core.Services.MetadataProviders;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.DataApiBuilder.Core.Resolvers.Factories
+{
+    /// <summary>
+    /// QueryEngineFactory class.
+    /// Used to get the appropriate queryEngine based on database type.
+    /// </summary>
+    public class QueryEngineFactory : IQueryEngineFactory
+    {
+        private readonly IEnumerable<IQueryEngine> _queryEngines;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QueryEngineFactory"/> class.
+        /// </summary>
+        /// <param name="runtimeConfigProvider">runtimeConfigProvider</param>
+        /// <param name="queryManagerFactory">queryManagerFactory</param>
+        /// <param name="metadataProviderFactory">metadataProviderFactory</param>
+        /// <param name="cosmosClientProvider">cosmsClientProvider.</param>
+        /// <param name="contextAccessor">HttpContextAccessor</param>
+        /// <param name="authorizationResolver">AuthorizationResolver.</param>
+        /// <param name="gQLFilterParser">gQLFilterParser</param>
+        /// <param name="logger">logger</param>
+        public QueryEngineFactory(RuntimeConfigProvider runtimeConfigProvider,
+            IQueryManagerFactory queryManagerFactory,
+            IMetadataProviderFactory metadataProviderFactory,
+            CosmosClientProvider cosmosClientProvider,
+            IHttpContextAccessor contextAccessor,
+            IAuthorizationResolver authorizationResolver,
+            GQLFilterParser gQLFilterParser,
+            ILogger<IQueryEngine> logger)
+        {
+            _queryEngines = new List<IQueryEngine>();
+
+            IEnumerable<DataSource> dataSources = runtimeConfigProvider.GetConfig().ListAllDataSources();
+
+            bool sqlEngineNeeded = dataSources.Any
+                (x => x.DatabaseType == DatabaseType.MSSQL || x.DatabaseType == DatabaseType.PostgreSQL || x.DatabaseType == DatabaseType.MySQL);
+
+            if (sqlEngineNeeded)
+            {
+                _queryEngines = _queryEngines.Append(new SqlQueryEngine(queryManagerFactory, metadataProviderFactory, contextAccessor, authorizationResolver, gQLFilterParser, logger, runtimeConfigProvider));
+            }
+
+            bool cosmosEngineNeeded = dataSources.Any
+                (x => x.DatabaseType == DatabaseType.CosmosDB_NoSQL || x.DatabaseType == DatabaseType.CosmosDB_PostgreSQL);
```

Comment:
Although we don't support cosmosdb_postgresql yet, but that would be a sql engine I guess.

Reply:
I havent included db's we are not supporting yet.

Reply:
But we have included it here, right?

Reply:
This is correct. i have removed it. should not support dbs that are not yet supported.